File: Cryptographic_Frontend/cf/SignInitMethod.cpp
#include "SignInitMethod.hpp"
#include "ResponseMessage.hpp"
#include "Argument.hpp"

cf::SignInitMethod::SignInitMethod(std::string mechanism, long long privateKeyHandler)
  : Method("SignInit")
{
  addArgument(ArgumentPtr(new StringArgument("mechanism", mechanism)));
  addArgument(ArgumentPtr(new IntegerArgument("handler", privateKeyHandler)));
}

cf::ResponseMessagePtr cf::SignInitMethod::parseResponse(const std::string & message) {
    ResponseMessagePtr response(new ResponseMessage());
    return std::move(response);
}
// kate: indent-mode cstyle; indent-width 4; replace-tabs on;

File: Cryptographic_Backend/src/main/java/cl/niclabs/cb/backend/methods/SignInitMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;

public interface SignInitMethod extends Method {
	public static class Args {
        public final String sessionHandler;
		public final String mechanism;
		public final String handler;

        public Args(String sessionHandler, String mechanism, String handler) {
            this.sessionHandler = sessionHandler;
            this.mechanism = mechanism;
            this.handler = handler;
        }
    }
}
